### DISCUSSION TITLE
Can beanie support dynamically creating a document and dynamically registering a document after init_beanie

### DISCUSSION THREAD

**Discussion by vasttian** [BeanieODM/beanie#559] [Question]
Can beanie support dynamically creating a document and dynamically registering a document after init_beanie?
Because I may have many documents, I want to register it when I need to use a certain document (similar to: document_models in init_beanie).

I might not be able to define the model in advance. Because the collection name is indeterminate(Or can beanie support defining a template model in advance? Then modify the name of the collection when creating a new one).

Example: 
A product has many projects. I need to automatically create a collection named project name in MongoDB to store the data of the current project every time I create a new project. Then initialize this collection to perform some database operations when you need to edit the project.

Thx.

**Comment by vasttian** [comment]
@roman-right Can you take a look for me? Thank you

**Comment by roman-right** [comment]
@vasttian Hi!
Sorry for the delay.

If you have the document class, you can init it at any moment in runtime using init_beanie. If you have a specific usecase where you can not use it, please explain it for me - I'll think how to support it

**Reply by vasttian** [reply]
hi,

Because I have many collections, I will create a new collection through the interface, and the name of the collection is dynamic.

If I define the document class in advance, and then initialize (init_beanie), I can create a new collection (if it does not exist).

May I ask how to dynamically create a new collection through init_beanie and specify the name (the name is specified by the user), and then initialize and operate the collection through beanie.
